src/components/Layout/Layout.tsx

```diff
@@ -0,0 +1,194 @@
+import {
+  FC, useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { useParams } from 'react-router-dom';
+import { TodoHeader } from '../TodoHeader';
+import { TodoList } from '../TodoList';
+import { TodoFooter } from '../TodoFooter';
+import { ErrorNotification } from '../ErrorNotification';
+import { Todo } from '../../types/Todo';
+import { TodoStatusFilter } from '../../types/TodoStatusFilter';
+import {
+  createTodo, deleteTodo, getTodos, updateTodo,
+} from '../../api/todos';
+import {
+  getActiveTodos, getCompletedTodos, getVisibleTodos, validateTitle,
+} from '../../utils/helpers';
+
+const USER_ID = 10684;
+
+export const Layout: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loadingTodos, setLoadingTodos] = useState<number[]>([0]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const { status } = useParams();
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((setTodos))
+      .catch((e) => setError(e.message));
+  }, []);
+
+  useEffect(() => {
+    let timerId: number;
+
+    if (error) {
+      timerId = window.setTimeout(() => {
+        setError(null);
+      }, 3000);
+    }
+
+    return () => window.clearTimeout(timerId);
+  }, [error]);
+
+  const completedTodos = useMemo(() => (
+    getCompletedTodos(todos)
+  ), [todos]);
+
+  const activeTodos = useMemo(() => (
+    getActiveTodos(todos)
+  ), [todos]);
+
+  const visibleTodos = useMemo(() => (
+    getVisibleTodos(todos, status as TodoStatusFilter)
+  ), [todos, status]);
+
+  const activeTodosLeft = activeTodos.length;
+
+  const hasTodos = todos.length > 0;
+  const hasVisibleTodos = todos.length > 0 || Boolean(tempTodo);
+  const isAllTodosCompleted = !activeTodos.length;
```

Comment:
```suggestion
  const hasActiveTodos = !activeTodos.length;
```

Reply:
@silchyn it's a great solution if our statement will be activeTodos.length > 0, but we can't even use `hasCompletedTodos`, because it means that we have at least one completed todo, and we need to make sure that all are completed